spec/unit/ledger_spec.rb

```diff
@@ -11,9 +11,7 @@
   end
 
   describe 'Initialisation' do
-    it 'Has a transactions array' do
-      expect(ledger.transactions).to be_a Array
-    end
+    it { is_expected.to respond_to :transactions }
```

Comment:
Hmmm interesting.  This is now testing the behaviour, which is great.  But generally, if an object allows another object to get its data, it's probably not encapsulated enough.  Look up the "tell, don't ask" idea.